File: matiasmasso/Mat.NET/DTO/DTOBaseGuidCodNom.cs
using System;

namespace DTO
{
    public class DTOBaseGuidCodNom : DTOBaseGuid
    {
        public Cods cod { get; set; }
        public string nom { get; set; }

        public enum Cods
        {
            notSet,
            vehicle,
            productBrand,
            productCategory,
            productSku,
            liniaTelefon
        }

        public DTOBaseGuidCodNom() : base()
        {
        }

        public DTOBaseGuidCodNom(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOBaseGuidCodNom Factory(Guid oGuid, Cods oCod = Cods.notSet, string sNom = "")
        {
            var retval = new DTOBaseGuidCodNom(oGuid);
            {
                var withBlock = retval;
                withBlock.cod = oCod;
                withBlock.nom = sNom;
            }
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOBaseGuidCodNom.vb
Public Class DTOBaseGuidCodNom
    Inherits DTOBaseGuid

    Property cod As Cods
    Property nom As String

    Public Enum Cods
        NotSet
        Vehicle
        ProductBrand
        ProductCategory
        ProductSku
        LiniaTelefon
    End Enum

    Public Sub New()
        MyBase.New
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function Factory(oGuid As Guid, Optional oCod As Cods = Cods.NotSet, Optional sNom As String = "") As DTOBaseGuidCodNom
        Dim retval = New DTOBaseGuidCodNom(oGuid)
        With retval
            .Cod = oCod
            .Nom = sNom
        End With
        Return retval
    End Function
End Class
